### ISSUE TITLE
disable WebSocket and IPC will cause kernel panick on exit 

### ISSUE THREAD

**Issue by MiaoShi09** [aionnetwork/aionr#26]
> os & version: ubuntu 16.04
> branch: dev_rf
> commit: 8fe44d7
---------------------------------------------------------------------------
**Pre-conditions:** disable websocket or ipc in command line( `--no-ws` or `--no-ipc`) or in configurations when launch aionr;

**Steps**:
1. When aionr starts syncing blocks or starts importing the blocks from miner, press CTRL+C

**Expected**: the kernel shut down properly.

**Actual**: the kernel panicked:
```bash
^C2019-06-11 14:23:29 Finishing work, please wait...

====================

stack backtrace:
   0:     0x55d0dc3bd8fc - backtrace::backtrace::trace::h61547d182043fd14
   1:     0x55d0dc3bc8b2 - <backtrace::capture::Backtrace as core::default::Default>::default::h78feaee0be4a90ff
   2:     0x55d0dc3bc928 - backtrace::capture::Backtrace::new::h2d6d4843fc894739
   3:     0x55d0dbe4922a - panic_hook::panic_hook::h1b02266b6ad6b97f

Thread 'main' panicked at 'Invalid IPC server instance!', libcore/option.rs:960

This is a bug. Please report it at:

    https://github.com/aionnetwork/aionr/issues/new
./custom.sh: line 3: 24440 Aborted                 (core dumped) ./aion --config=custom/custom.toml $*
```
```bash
==================== 

  

stack backtrace: 

   0:     0x55c7279068fc - backtrace::backtrace::trace::h61547d182043fd14 

   1:     0x55c7279058b2 - <backtrace::capture::Backtrace as core::default::Default>::default::h78feaee0be4a90ff 

   2:     0x55c727905928 - backtrace::capture::Backtrace::new::h2d6d4843fc894739 

   3:     0x55c72739222a - panic_hook::panic_hook::h1b02266b6ad6b97f 

  

Thread 'main' panicked at 'Invalid WS server instance!', libcore/option.rs:960 

  

This is a bug. Please report it at: 

  

    https://github.com/aionnetwork/aionr/issues/new 

./mastery.sh: line 3: 12748 Aborted                 (core dumped) ./aion --config=mastery/mastery.toml $* 
```

**Comment by MiaoShi09** [comment]
HTTP also has the same problem:
```bash
^C2019-06-11 14:40:20 Finishing work, please wait...

====================

stack backtrace:
   0:     0x5562cb9628fc - backtrace::backtrace::trace::h61547d182043fd14
   1:     0x5562cb9618b2 - <backtrace::capture::Backtrace as core::default::Default>::default::h78feaee0be4a90ff
   2:     0x5562cb961928 - backtrace::capture::Backtrace::new::h2d6d4843fc894739
   3:     0x5562cb3ee22a - panic_hook::panic_hook::h1b02266b6ad6b97f

Thread 'main' panicked at 'Invalid HTTP server instance!', libcore/option.rs:960

This is a bug. Please report it at:

    https://github.com/aionnetwork/aionr/issues/new
./custom.sh: line 3: 24808 Aborted                 (core dumped) ./aion --config=custom/custom.toml $*

```

**Comment by aion-chris** [comment]
fixed at dc86ca94ea6674879457d72fd6e6f0f7adafe78a
